shared/src/main/kotlin/com/egm/stellio/shared/util/JsonLdUtils.kt

```diff
@@ -285,3 +295,22 @@ fun String.extractShortTypeFromExpanded(): String =
      * TODO do a clean implementation using info from @context
      */
     this.substringAfterLast("/").substringAfterLast("#")
+
+fun Map<String, Any>.toKeyValues(): Map<String, Any> {
```

Comment:
I'm a bit worried about this function because it can be called on any map, whereas it will only work when called from a compacted entity.

I would be in favor of some type safety :)

For instance, `JsonLdEntity.compact()` could return a `CompactedJsonLdEntity` (maybe just a `typealias` on `Map<String, Any>` for the moment). Then have a `CompactedJsonLdEntity.toKeyValues()`.

Reply:
me too, I was not satisfied by this suggested extension function. I think extensions functions when extending standard classes should do a generic work, not a specific one 
for example String.toSqlColumnName() me is too specific for the choice of implementation

Reply:
with typealias the extension function and its use seem better